src/compose/table.c

```diff
@@ -84,27 +102,12 @@ xkb_compose_table_new_from_file(struct xkb_context *ctx,
                                 enum xkb_compose_format format,
                                 enum xkb_compose_compile_flags flags)
 {
-    struct xkb_compose_table *table;
-    bool ok;
-
-    if (flags & ~(XKB_COMPOSE_COMPILE_NO_FLAGS)) {
-        log_err_func(ctx, XKB_LOG_MESSAGE_NO_ID,
-                     "unrecognized flags: %#x\n", flags);
-        return NULL;
-    }
-
-    if (format != XKB_COMPOSE_FORMAT_TEXT_V1) {
-        log_err_func(ctx, XKB_LOG_MESSAGE_NO_ID,
-                     "unsupported compose format: %d\n", format);
-        return NULL;
-    }
-
-    table = xkb_compose_table_new(ctx, locale, format, flags);
+    struct xkb_compose_table * const table =
```

Comment:
is that `const` here intended? It's not technically incorrect, just ... unusual

Reply:
It is intended: I started to qualify every variable as `const` whenever relevant. I find that it helps a bit: better semantics, specific IDE highlighting, distinguish argument pointers from local pointers…